block/block_service.proto

```diff
@@ -539,15 +517,41 @@ message SubscribeStreamResponse {
         SubscribeStreamResponseCode status = 1;
 
         /**
-         * A stream response item containing a single `BlockItem`.
+         * A stream response item containing one or more `BlockItem`s.
          * <p>
-         * The full response SHALL consist of many `block_item` messages
+         * The full response SHALL consist of many `block_items` messages
          * followed by a single `status` message.
          */
-        com.hedera.hapi.block.stream.BlockItem block_item = 2;
+        SubscribeStreamResponseSet block_items = 2;
     }
 }
 
+/**
+ * A wrapper message to permit repeated block items in a response.<br/>
+ * This message is required so that we can return ordered lists of block
+ * items to a subscriber.
+ *
+ * Each `SubscribeStreamResponseSet` MUST contain at least one `BlockItem`,
+ * and MAY contain up to one full block.<br/>
+ * A single `SubscribeStreamResponseSet` SHALL NOT contain block items from
+ * more than one block.<br/>
+ * If a `BlockHeader` is present in a `SubscribeStreamResponseSet`, that item
+ * MUST be the first item in the list.<br/>
+ * If a `BlockProof` is present in a `SubscribeStreamResponseSet`, that item
+ * MUST be the last item in the list.
+ */
+message SubscribeStreamResponseSet {
+    /**
+     * An ordered list of `BlockItem`s.<br/>
+     * This list supports sending block items to subscribers in batches
+     * for greater channel efficiency.
+     * <p>
+     * The full response SHALL consist of many of these collections
+     * followed by a single `status` message.
+     */
+    repeated com.hedera.hapi.block.stream.BlockItem block_items = 1;
+}
```

Comment:
As is my understanding we are only going to send a "Response" once a whole block is received. Am I right?
The why we need to send the whole block_items back as part of the Response, it would not be more effective to have a response object that is only a block number instead of streaming back all the items again?

Reply:
We are not guaranteed to send only whole blocks back, as the low-latency unverified stream requires data to flow through immediately.

Reply:
I see, I was mixing the clients 😅 
Thank you!